doc/build/snippets/using.rst

```diff
@@ -0,0 +1,39 @@
+.. _using-snippets:
+
+Using Snippets
+##############
+
+.. tip::
+
+   See :ref:`built-in-snippets` for a list of snippets that are provided by
+   Zephyr.
+
+Snippets have names. You use snippets by giving their names to the build
+system.
+
```

Comment:
>  even if you re-run cmake with different snippets, that argument will be completely ignored 

Nice catch. I think the right thing to do here is compare CACHED_SNIPPET with the value of the snippets command line argument, and:

1. do a pristine build if `build.pristine` is `auto`
2. complain and error out otherwise

Seem right to you?

Reply:
Yep, works for me

Reply:
> Yep, works for me

@nordicjm I got cold feet about this idea the more I thought about it. I think it might be better to just force a cmake re-run if `-S` is given. This is more similar to the behavior documented in https://docs.zephyrproject.org/latest/develop/west/build-flash-debug.html#one-time-cmake-arguments for when you provide one-time CMake arguments... which is exactly what we're doing here.